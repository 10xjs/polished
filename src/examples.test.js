#!/bin/env node

/**
 * This file recursively tests all .examples.js files in the src/ folder and snapshots them with
 * Jest.
 *
 * NOTE: Changing anything about the test names breaks all previous snapshots. Handle with care.
 */

const path = require('path')
const { getExamples, getName } = require('../helpers')

getExamples('./').forEach((exampleFile) => {
  describe(`${getName(exampleFile)} examples`, () => {
    // eslint-disable-next-line global-require, import/no-dynamic-require
    const examples = require(path.resolve('./', exampleFile))
    examples.forEach((example, index) => {
      it(`should match example #${index}`, () => {
        expect(example).toMatchSnapshot()
      })
    })
  })
})
