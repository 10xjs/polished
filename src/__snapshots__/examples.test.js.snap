exports[`stripUnit examples should match example #0 1`] = `
Object {
  "--dimension": 100,
}
`;

exports[`stripUnit examples should match example #0 2`] = `
Object {
  "--dimension": 100,
}
`;

exports[`stripUnit examples should match example #1 1`] = `
"
  --dimension: 100;
"
`;

exports[`stripUnit examples should match example #1 2`] = `
"
  --dimension: 100;
"
`;
